lib/ir_pagerduty.rb

```diff
@@ -0,0 +1,19 @@
+require 'pagerduty/full'
+
+class IRPagerduty < PagerDuty::Full
+  def initialize
+    @apikey = SupportApp.pager_duty_token
+    @subdomain = SupportApp.pager_duty_subdomain
+  end
+
+  def fetch_json(path, params)
+    res = self.api_call(path, params)
+
+    case res
+      when Net::HTTPSuccess
+        JSON.parse(res.body)
+      else
+        res.error!
+    end
+  end
```

Comment:
I was thinking specifically about the duplication between different classes that speak to PagerDuty.

But I think I mis-read the diff as you're now using this class elsewhere anyway. So nope, this is fine as it is.